source/_cookbook/SSL guide.md

```diff
@@ -0,0 +1,570 @@
+
+Duckdns.org will now generate personalised instructions for you to follow so that your device can update their website every time your IP address changes.  Carefully follow the instructions given on duckdns.org to set up your device.
+
+At the end of the instructions DuckDNS will suggest you set up port forwarding.  No need, we have already done this in step 2.
+
+What you have now done is set up DuckDNS so that whenever you type examplehome.duckdns.org in to your browser it will convert that to your router's external IP address.  Your external IP address will always be up to date because your device running HA will update DuckDNS every time it changes.
+
+Now type your new URL in to your address bar on your browser with port 8123 on the end:
+
+```text
+http://examplehome.duckdns.org:8123
+```
+
+What now happens behind the scenes is this:
+
+*DuckDNS receives the request and forwards the request to your router's external IP address (which has been kept up to date by your device running HA)
+*Your router receives the request on port 8123 and checks the port forwarding rules
+*It finds the rule you created in step 2 and forwards the request to your HA instance
+*Your browser displays your HA instance frontend.
+
+Did it work? Super!
+
+You now have a remotely accesible HA instance that has a text-based URL and will not drop out if your service provider changes your IP.  But, it is only as secure as the password you set, which can be snooped during your session by a malicious hacker with relative ease.  So we need to set up some encryption with SSL, read on to find out how.
+
+#### {% linkable_title 4 - Obtain an ssl certificate from LetsEncrypt %}
+
+First we need to set up another port forward like we did in step 2.  Set your new rule to:  
+
+```text
+Service name - ha_letsencrypt
+Port Range - 80
+Local IP - YOUR-HA-IP
+Local Port - 80
+Protocol - Both
+```
+
+Remember to save the new rule.
+
+<p class='note'>
+In cases where your ISP blocks port 80 you will need to change the port forward options to forward port 443 from outside to port 443 on your HA device.  Please note that this will limit your options for automatically renewing the certificate, but this is a limitation because of your ISP setup and there is not a lot we can do about it!
+</p>
+
+now SSH in to the device your HA is running on.
+
+
+<p class='note'>
+If you're running the 'standard' setup on a raspberry pi the chances are you just logged in as the 'pi' user.  If not, you may have logged in as the HA user.  There are commands below that require the HA user to be on the sudoers list.  If you are not using the 'standard' pi setup it is presumed you will know how to get your HA user on the sudoers list before continuing.  If you are running the 'standard' pi setup, from your 'pi' user issue the following command (where <hass> is the HA user):

```

Comment:
Technical Question: Why does the user need to be on the sudoers list? I think it's not a good idea to give homeassistant that priviledge, since it's running the web services.
EDIT: Ah certbot needs root in --standalone mode. Bummer.